src/charm.py

```diff
@@ -217,6 +254,13 @@ def _pebble_layer(self) -> Layer:
     def _domain_url(self) -> Optional[str]:
         return self.config["external_url"] or self.public_ingress.url
 
+    @property
+    def _log_level(self) -> str:
+        level = self.config["log_level"]
+        if level not in LOG_LEVELS:
+            return DEFAULT_LOG_LEVEL
```

Comment:
Not sure if we want this. At the very least if the user has provided an incorrect log level we need to warn them about it.

Reply:
Agreed, it's fine to go in blocked state if the user config is not correct. We don't want to encourage/tolerate writing bad configurations.